File: src/main/java/com/websocket/demo/room/ChatRoom.java
package com.websocket.demo.room;

import lombok.*;

import java.time.LocalDateTime;

@Getter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public final class ChatRoom {
    private String id;
    private String title;
    private LocalDateTime lastModifiedAt;
    private LocalDateTime createdAt;

    /**
     * for redis deserialize
     */
    public ChatRoom(String id) {
        this.id = id;
    }
}

File: src/main/java/com/websocket/demo/room/ChatRoom.kt
package com.websocket.demo.room

import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "chat_room")
class ChatRoom(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    var title: String,
    var lastModifiedAt: LocalDateTime?,
    var createdAt: LocalDateTime?
) {
    constructor(): this(0, "", null,  LocalDateTime.now())

    constructor(title: String): this(null, title, null, LocalDateTime.now())
}